core/src/com/unciv/models/ruleset/unit/BaseUnitCost.kt

```diff
@@ -11,6 +11,13 @@ class BaseUnitCost(val baseUnit: BaseUnit) {
 
     fun getProductionCost(civInfo: Civilization): Int {
         var productionCost = baseUnit.cost.toFloat()
+
+        for (unique in baseUnit.getMatchingUniques(UniqueType.CostIncreasesPerCity, StateForConditionals(civInfo)))
```

Comment:
Sounds to me like we really need a shared function in a parent class

Reply:
You mean like `IConstruction.getProductionCost`? Shared function exists, but is blank rn

Reply:
Looking over this again the shared code is like 7 lines in total, I think it's fine to leave as-is.